### PR TITLE
Director fully async

### LINKED ISSUE #297: make director calls to docker async
When starting a service the director waits for it be ready synchronously which blocks other calls (for example listing services)

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] replace calls to official python docker SDK by aiodocker (wraps official docker REST API)
Bonus:
- [x] dynamic services are limited to 4CPUs and 16GB by default unless set otherwise
- [x] dynamic services are restarted 2 times in case of failures

## Related issue number

fixes #297 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/director/src/simcore_service_director/producer.py [hunk 1]
```diff
@@ -135,118 +202,30 @@ def _get_swarm_network(client: DockerClient) -> docker.models.networks.Network:
...
-
-def __set_service_name(docker_service_runtime_parameters: Dict,
-                       service_name: str,
-                       node_uuid: str):
-    # pylint: disable=C0103
-    docker_service_runtime_parameters["name"] = service_name + "_" + node_uuid
-    log.debug("Added service name parameter to docker runtime parameters: %s",
-              docker_service_runtime_parameters["name"])
-
+async def _get_swarm_network(client: aiodocker.docker.Docker) -> Dict:
```
Comment (pcrespov) [inline services/director/src/simcore_service_director/producer.py]: I am surprised that python likes the fact that you are overloading ``_get_swarm_network`` (one sync and the other async). I would rename them.

File: services/director/requirements/_base.txt [hunk 2]
```diff
@@ -4,22 +4,22 @@
 #
 #    pip-compile --output-file=_base.txt _base.in
 #
+aiodocker==0.14.0
 aiohttp==3.3.2
 git+https://github.com/ITISFoundation/aiohttp_apiset.git@fixes_4_osparc#egg=aiohttp_apiset
+async-generator==1.10     # via asyncio-extras
 async-timeout==3.0.1      # via aiohttp
+asyncio-extras==1.3.2
```
Comment (pcrespov) [inline services/director/requirements/_base.txt]: we should perhaps update to python 3.7 sometime soon :-)
Async context managers, e.g. are already there https://docs.python.org/3/library/contextlib.html

File: services/director/src/simcore_service_director/producer.py (hunk 1 above)
Comment (sanderegg) [reply services/director/src/simcore_service_director/producer.py]: it does not like it...
it was a bad merge

File: services/director/requirements/_base.txt (hunk 2 above)
Comment (sanderegg) [reply services/director/requirements/_base.txt]: agree

**Pull request merged**